### PR TITLE
Align API version of components with API version in sfdx-project.json

### PR DESCRIPTION
**Pull request by OlivierBrabanders**

### CODE REVIEW

**Comment by davidbrowaeys** [comment]
Very good 2 things to add : 

- We need to be able to filter by metadata type only. So maybe we can add something like  --metadatatype param. 
example--m "ApexClass". If flag is not present then we need to process for all. 

- Sometimes, we want to be able to specify a api version on a specific class. So we need to maintain an exclusion list so that it will never process those. To do so I am proposing to add a new attribute as apiAlignmentExclusion under dxb in the sfdx-plugin
example: 
`"apiAlignmentExclusion": ["force-app/main/default/classes/MyBadClass.cls-meta.xml","force-app/main/default/classes/SuperbadClass.cls-meta.xml"]`

**Review by davidbrowaeys — CHANGES_REQUESTED** [review]
Very good 2 things to add :

We need to be able to filter by metadata type only. So maybe we can add something like --metadatatype param.
example--m "ApexClass". If flag is not present then we need to process for all.

Sometimes, we want to be able to specify a api version on a specific class. So we need to maintain an exclusion list so that it will never process those. To do so I am proposing to add a new attribute as apiAlignmentExclusion under dxb in the sfdx-plugin
example:
"apiAlignmentExclusion": ["force-app/main/default/classes/MyBadClass.cls-meta.xml","force-app/main/default/classes/SuperbadClass.cls-meta.xml"]

File: src/commands/dxb/api/align.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    `$ sfdx dxb api align --metadata-type ApexClass --metadata-type ApexTrigger`
+  ];
+
+  public static args = [{ name: 'file' }];
+
+  protected static flagsConfig = {
+    'metadata-type': flags.string({
+      char: 'm',
+      description: 'Select specific metadata type to align, value is the name of the root tag of the XML file holding the apiVersion tag i.e. <ApexClass ...',
+      multiple: true
```
Comment (davidbrowaeys) [inline src/commands/dxb/api/align.ts:25]: nice

**Comment by OlivierBrabanders** [comment]
Ready for review David!

**Pull request merged**